src/model.cpp

```diff
@@ -175,7 +175,11 @@ Status Model::retireVersions(std::shared_ptr<model_versions_t> versionsToRetire)
         }
         cleanupModelTmpFiles(modelVersion->getModelConfig());
         updateDefaultVersion(version);
-        modelVersion->unloadModel();
+        if (failedVersions.find(version) != failedVersions.end()) {
```

Comment:
In line 165 you get:

Will unload model: {}

However you may just cleanup model files (not retire model). I think we need to split those two calls at the Model class level as well.